lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -2094,6 +2099,85 @@ void PullbackCloner::Implementation::visitSILBasicBlock(SILBasicBlock *bb) {
       auto *predBB = std::get<0>(pair);
       auto incomingValue = std::get<1>(pair);
       blockTemporaries[getPullbackBlock(predBB)].insert(concreteBBArgAdjCopy);
+      // Handle `switch_enum` on `Optional`.
+      if (auto *termInst = bbArg->getSingleTerminator()) {
+        if (auto *sei = dyn_cast<SwitchEnumInst>(termInst)) {
+          auto *optionalEnumDecl = getASTContext().getOptionalDecl();
+          auto operandTy = sei->getOperand()->getType();
+          if (operandTy.getASTType().getEnumOrBoundGenericEnum() == optionalEnumDecl) {
+            // `Optional<T>`
+            auto optionalTy = remapType(operandTy);
+            // `T`
+            auto Ty = optionalTy.getOptionalObjectType();
+            // `T.TangentVector`
+            auto tanTy = remapType(concreteBBArgAdjCopy->getType());
+            // `Optional<T.TangentVector>`
+            auto optionalTanTy = SILType::getOptionalType(tanTy);
+            // `Optional<T>.TangentVector`
+            auto tangentVectorTy = getRemappedTangentType(optionalTy);
```

Comment:
The goal here seems to be to get the `Optional<T>.TangentVector` type. Could you collapse these into a single call to `getRemappedTangentType`?

```suggestion
            // `Optional<T>.TangentVector`
            auto optionalTanTy = getRemappedTangentType(operandTy);
```

Reply:
The goal is to get two things:
1. `Optional<T>.TangentVector`
2. `Optional<T.TangentVector>` (for a parameter to pass to`Optional<T>.TangentVector.init`)
The `getRemappedTangentType` gives 1., but I don't see a straightforward way to shorten the rest of the code that gets 2.

Reply:
Ah, I see. Both types seem necessary.